### DISCUSSION TITLE
Option for slugs to be non-translatable

### DISCUSSION THREAD

**Discussion by ItagMichael** [craftcms/cms#8962] [Ideas] [answered]
### Description

Would be great to be able to specify if a slug for an entry should or should not be translatable across sites.

Use case is a situation where you want to have certain entries propagated across multiple sites with identical content. #2856 made this easier with the  _Title Translation Method_ setting, but currently the slug must still be changed individually per site even if an editor wanted them to be the same .

This could perhaps be related to the _Title Translation Method_ options in an entry type's settings, or configured via a _Slug Translation Method_ dropdown.

**Comment by domstubbs** [comment]
A hearty 👍 from me, as it feels like an oversight that I can designate all fields, including the title, as non-translatable but the slug field is the one exception. In situations where you want 1:1 shared content across sites it’s unhelpful to have this odd caveat to mention to clients, particularly as slugs are something that they tend not to give much (or any) thought to in the first place.

**Comment by MoritzLost** [comment]
I agree with this suggestion as well. There are a couple of use-cases for keeping slugs in sync between sites. In particular, we're currently building a headless data management system, where sites are used for translation into multiple languages. The slug is used by the external system consuming the data as unique identifiers. Because of this, the slugs need to be kept in sync.

The ideal solution would be an additional setting `Slug translation method` similar to the `Title translation method`. Just tying the slug to the `Title translation method` is not a good solution in my opinion. There are many use-cases (this one included) were we _do_ want the title do be translated, but _don't_ want the slug to be translated.

Does somebody have a working hook that propagates slug changes to all sites on save?

**Comment by abhi44mbd** [comment]
Anyone knows how can I implement this prevent slug translations in the translations plugin.?

**Comment by brandonkelly** [comment, marked as answer]
Craft 4.5.0 is out now, with a new “Slug Translation Method” setting on entry types, so you can control slugs’ translatable behavior just like with titles and custom fields.

**Reply by MoritzLost** [reply]
That's a great improvement, thanks!
